core/includes/install.inc

```diff
@@ -722,6 +723,11 @@ function backdrop_rewrite_settings($settings = array()) {
         preg_match('/\$([^ =]*) /', $line, $variable);
         $variable_name = $variable[1];
         if (array_key_exists($variable_name, $settings)) {
+          // If variable_name is 'database', we need to remove the next 7
+          // lines as they are part of the default database array.
+          if ($variable_name == 'database') {
+            $database_line_count = 7;
```

Comment:
This relies on the expectation that database definition is left to the default:
```php
$database = array(
  'driver' => 'mysql',
  'database' => 'database_name',
  'username' => 'user',
  'password' => 'pass',
  'host' => 'localhost',
  'prefix' => '',
);
```

What if someone removes the prefix line or formats the entire array to be in a single line? I believe that we need to use the `m` (multiline) operator in `preg_match()` and adjust the regex accordingly to be checking several lines until it explicitly reaches a `);`.

Reply:
@klonos - as the undisputed king of regex, please could I ask you to make a code suggestion that I can accept here?

Reply:
Resolving this as per discussions at Backdrop developer meetings